examples/lang/erc721/src/lib.rs

```diff
@@ -0,0 +1,230 @@
+// Copyright 2018-2019 Parity Technologies (UK) Ltd.
+// This file is part of ink!.
+//
+// ink! is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// ink! is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License
+// along with ink!.  If not, see <http://www.gnu.org/licenses/>.
+
+#![cfg_attr(not(feature = "std"), no_std)]
+
+use ink_core::{
+    env::DefaultSrmlTypes,
+    memory::format,
+    storage,
+};
+use ink_lang::contract;
+use ink_model::EnvHandler;
+use ink_core::storage::alloc::Initialize as _;
+use ink_core::storage::alloc::AllocateUsing as _;
+use std::ops::Index;
+
+pub type Result<T, E> = core::result::Result<T, E>;
+
+contract! {
+    #![env = DefaultSrmlTypes]
+
+    // Event deposited when a token transfer occurs
+    event Transfer {
+        from: AccountId,
+        to: AccountId,
+        token_id: u32,
+    }
+
+    // Event deposited when an approval occurs
+    event Approval {
+        owner: AccountId,
+        to: AccountId,
+        token_id: u32,
+    }
+
+    /// The storage items for a typical ERC721 token implementation.
+    struct Erc721 {
+        /// The total supply.
+        token_owner: storage::HashMap<u32, AccountId>,
+        token_approvals: storage::HashMap<u32, AccountId>,
+        owned_tokens: storage::HashMap<AccountId, storage::Vec<u32>>,
+        all_tokens: storage::Vec<u32>,
+        name: storage::Vec<u8>,
+        symbol: storage::Vec<u8>,
+
+    }
+
+    impl Deploy for Erc721 {
+        fn deploy(&mut self, name: u8, symbol: u8) {
+            self.name.push(name);
+            self.symbol.push(symbol);
+        }
+    }
+
+    impl Erc721 {
+
+        pub(external) fn get_total_supply(&self) -> u32 {
```

Comment:
Is this part of the "official" Erc721 standard? I am missing it here: https://github.com/OpenZeppelin/openzeppelin-contracts/blob/master/contracts/token/ERC721/IERC721.sol Maybe I am looking wrong.

Reply:
https://github.com/eduadiez/DAppNodeNFT/blob/master/contracts/ERC721.sol

Reply:
Then this implementation seems to me to be a superset of the Erc721 standard.